frontend/services/template.service.ts

```diff
@@ -0,0 +1,41 @@
+import { useFetch } from '#app';
+import type { TemplateSchema, FormData } from '@/types/template';
+
+/**
+ * Service for handling template-related API operations
+ */
+class TemplateService {
+    private readonly baseUrl = 'http://localhost:5052/api/templates';
```

Comment:
Avoid harcoding the API url here, especially as with the `localhost` hostname it will not work on deployed code.
Instead you could either:
- use a configuration file to configure the API hostname, like `.env` for local development and [static web app environment variables](https://learn.microsoft.com/en-us/azure/static-web-apps/application-settings) for deployed code
- remove the hostname from the endpoint (i.e. make the requests directly to `/api`), the linked backend should take care of the redirection automatically on the deployed app and for local development you could configure a devProxy in the Nuxt configuration.